DisCatSharp.ApplicationCommands/Workers/ApplicationCommandWorker.cs

```diff
@@ -0,0 +1,395 @@
+// This file is part of the DisCatSharp project, a fork of DSharpPlus.
+//
+// Copyright (c) 2021 AITSYS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using DisCatSharp.Entities;
+using DisCatSharp.Enums;
+
+namespace DisCatSharp.ApplicationCommands
+{
+    /// <summary>
+    /// Represents a <see cref="CommandWorker"/>.
+    /// </summary>
+    internal class CommandWorker
+    {
+        /// <summary>
+        /// Parses context menu application commands.
+        /// </summary>
+        /// <param name="type">The type.</param>
+        /// <param name="methods">List of method infos.</param>
+        /// <param name="guildid">The optional guild id.</param>
+        /// <param name="translator">The optional command translations.</param>
+        /// <returns>Too much.</returns>
+        internal static Task<Tuple<List<DiscordApplicationCommand>, List<KeyValuePair<Type, Type>>, List<ContextMenuCommand>>> ParseContextMenuCommands(Type type, IEnumerable<MethodInfo> methods, ulong? guildid = null, List<CommandTranslator> translator = null)
+        {
+            List<DiscordApplicationCommand> Commands = new();
+            List<KeyValuePair<Type, Type>> CommandTypeSources = new();
+            List<ContextMenuCommand> contextMenuCommands = new();
+
+
+            foreach (var contextMethod in methods)
+            {
+                var contextAttribute = contextMethod.GetCustomAttribute<ContextMenuAttribute>();
+
+                DiscordApplicationCommandLocalization NameLocalizations = null;
+
+                if (translator != null)
+                {
+                    var command_translation = translator.Single(c => c.Name == contextAttribute.Name && c.Type == contextAttribute.Type);
```

Comment:
Could reduce the nesting here by using null propagation on translator 

eg. `var command_translation = translator?.Single(c => c.Name == contextAttribute.Name && c.Type == contextAttribute.Type);`

That would get rid of the need for the first if block `if (translator != null)`